apps/next-evans/components/pages/home/news-feed/news-feed.ts

```diff
@@ -0,0 +1,507 @@
+import { t } from '@friends-library/locale';
+import { htmlShortTitle } from '@friends-library/adoc-utils';
+import type { Lang } from '@friends-library/types';
+import type { Audiobook, DocumentWithMeta, NewsFeedType } from '@/lib/types';
+import { months } from '@/lib/dates';
+import { LANG } from '@/lib/env';
+import { isNotNull } from '@/lib/utils';
+import { getDocumentUrl } from '@/lib/friend';
+
+export interface FeedItem {
+  month: string;
+  day: string;
+  year: string;
+  type: NewsFeedType;
+  title: string;
+  description: string;
+  url: string;
+  date: string;
+}
+
+type NewsfeedDocumentProps = Pick<
+  DocumentWithMeta,
+  | 'authorName'
+  | 'dateAdded'
+  | 'id'
+  | 'title'
+  | 'authorSlug'
+  | 'slug'
+  | 'editions'
+  | 'mostModernEdition'
+  | 'altLanguageId'
+>;
+
+export async function getNewsFeedItems(
+  lang: Lang,
+  documents: {
+    en: Array<NewsfeedDocumentProps>;
+    es: Array<NewsfeedDocumentProps>;
+  },
+): Promise<FeedItem[]> {
+  const outOfBandEvents = getOutOfBandEvents();
+  const recentAdditions = getRecentAdditions(`es`, documents.es, documents.en).concat(
+    LANG === `en` ? getRecentAdditions(`en`, documents.en) : [],
+  );
+  const recentAudios = getRecentAudios(documents[lang]);
+  return recentAdditions
+    .concat(outOfBandEvents.filter((item) => item.lang.includes(lang)))
+    .concat(recentAudios)
+    .sort((a, b) => new Date(b.date).getTime() - new Date(a.date).getTime())
+    .slice(0, MAX_NUM_NEWS_FEED_ITEMS);
+}
+
+function getRecentAdditions(
+  lang: Lang,
+  documents: Array<NewsfeedDocumentProps>,
+  altLanguageDocuments: Array<NewsfeedDocumentProps> = [],
+): FeedItem[] {
+  const type = LANG === `en` && lang === `es` ? `spanish_translation` : `book`;
+  return documents
+    .sort((a, b) => new Date(b.dateAdded).getTime() - new Date(a.dateAdded).getTime())
```

Comment:
what do you think about making some sort of lib util helper fn called `newestFirst` so we could clean up some of these noisy sorts to just read:

```tsx
  .sort((a, b) => newestFirst(a.date, b.date))
```